Library/Homebrew/cmd/audit.rb

```diff
@@ -676,6 +676,10 @@ def audit_text
     if text =~ /def plist/ && text !~ /plist_options/
       problem "Please set plist_options when using a formula-defined plist."
     end
+
+    if text =~ %r{require "language/go"} && text !~ /go_resource/
```

Comment:
Heh, it's funny you mention this because we had almost the exact same discussion a fair while ago now. I'll hollowly echo my comments from then which were more or less that the idea sounds good but I'd prefer not to conflate that change with this PR. Less hollowly, I'm happy to push a fresh PR afterwards.